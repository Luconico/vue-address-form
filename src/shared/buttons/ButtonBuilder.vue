<template>
      <button
        :type="type"
        :class="customClass"
        :disabled="disabled"
    >
      <LoaderSpinner v-if="isFetching" color="gray" size="1rem" />{{  buttonText  }}
    </button>
</template>

<script>
import LoaderSpinner from "@/shared/loader-spinner/LoaderSpinner.vue";
export default {
    name: "button-builder",
    components: {
        LoaderSpinner
    },
    props: {
        type: {
            type: String,
            default: "button"
        },
        buttonText: {
            type: String,
            default: "submit"
        },
        disabled: {
            type: Boolean,
            default: false
        },
        isFetching: {
            type: Boolean,
            default: false
        },
        customClass: {
            type: Object,
            default: () => ({})
        }
    }
}
</script>

<style>

</style>