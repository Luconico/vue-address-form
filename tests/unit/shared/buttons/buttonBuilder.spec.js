import { shallowMount } from '@vue/test-utils';
import ButtonBuilder from '@/shared/buttons/ButtonBuilder.vue';

describe('ButtonBuilder', () => {

    const wrapper = shallowMount(ButtonBuilder, {
        props: {
            type: 'submit',
            buttonText: 'test',
            customClass: { 'btn test-btn': true },
        }
    });

    it('should render the component', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('should match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should props type, buttonText and customClass', () => {
        expect(wrapper.props()).toEqual({
            type: 'submit',
            buttonText: 'test',
            customClass: { 'btn test-btn': true },
            disabled: false,
            isSubmitting: false,
        });
    });

});