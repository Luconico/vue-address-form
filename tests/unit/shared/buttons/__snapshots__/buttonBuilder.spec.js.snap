// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBuilder should match with snapshot 1`] = `
<button type="submit" class="btn test-btn">
  <!--v-if-->test
</button>
`;
